sky/core.py

```diff
@@ -551,10 +555,13 @@ def cancel(cluster_name: str,
         cluster_name, operation_str='Cancelling jobs')
 
     # Check the status of the cluster.
-    handle = backend_utils.check_cluster_available(
-        cluster_name,
-        operation='cancelling jobs',
-    )
+    if not _ignore_cluster_aliveness:
+        handle = backend_utils.check_cluster_available(
+            cluster_name,
+            operation='cancelling jobs',
+        )
+    else:
+        _, handle = backend_utils.refresh_cluster_status_handle(cluster_name)
```

Comment:
Can this return None, if the entire spot cluster is preempted?

Reply:
Good point! Just changed the implementation to make it more robust.